packages/skygear-core/lib/cloud/logging.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2015 Oursky Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import bunyan from 'bunyan';
+import _ from 'lodash';
+import { settings } from './settings';
+
+function normalizeLogFields(rec) {
+  /* eslint-disable no-unused-vars */
+  const {
+    name,
+    v,
+    hostname,
+    pid,
+    err,
+    level,
+    ...fields
+  } = rec;
+  /* eslint-enable no-unused-vars */
+
+  const newRec = {
+    logger: name,
+    level: bunyan.nameFromLevel[level],
+    process: 'node',
+    ...fields
+  };
+  if (err) {
+    newRec.error = err;
+  }
+  return newRec;
+}
+
+class TextStream {
+  write(rec) {
+    /* eslint-disable no-unused-vars */
+    const {
+      logger,
+      msg,
+      level,
+      time,
+      error,
+      tag,
+      ...fields
+    } = normalizeLogFields(rec);
+    /* eslint-enable no-unused-vars */
+    if (!_.isEmpty(fields)) {
+      console.log(`[${level}] ${tag}: ${msg}, ${JSON.stringify(fields)}`);
+    } else {
+      console.log(`[${level}] ${tag}: ${msg}`);
+    }
+    if (error) {
+      console.error(error.stack);
```

Comment:
I think it should either
- the log level to `console` should follow the incoming log level from the logger, or 
- all log to `console.log`. 

And I prefer the first one.
And same should apply to `JSONStream`.